File: src/models/phone_operator.js

class PhoneOperator {
    static MTN = "MTN Cameroon";
    static ORANGE = "Orange";
    static NEXTTEL = "NEXTTEL";
    static UNKNOWN = "UNKNOWN";
    
    static fromString(phoneOperator = '') {

        if(phoneOperator.startsWith("o") || phoneOperator.startsWith('O')){
            return PhoneOperator.ORANGE;
        }
        if(phoneOperator.startsWith("m") || phoneOperator.startsWith('M')){
            return PhoneOperator.MTN;
        }
        if(phoneOperator.startsWith("n") || phoneOperator.startsWith('n')){
            return PhoneOperator.NEXTTEL;
        }
        return PhoneOperator.UNKNOWN;
    }
}

module.exports = { PhoneOperator }

File: src/main/java/com/bixterprise/gateway/utils/PhoneOperator.java

package com.bixterprise.gateway.utils;

/**
 * This enum define the set of Known phone operator, Actually only Cameroon's is known
 * MTN -> MTN Cameroon
 * ORANGE -> Orange
 * NEXTTEL -> NEXTTEL
 * UNKNOWN -> UNKNOWN
 * @author haranov
 */
public enum PhoneOperator {
    
    MTN("MTN"), ORANGE("Orange"), NEXTTEL("NEXTTEL"), UNKNOWN("UNKNOWN");
    
    String name;
    
    PhoneOperator(String name){
        this.name = name;
    }
    
    @Override
    public String toString(){return this.name;}

}
